### PR TITLE
BAVL-71 adding the necessary client for making calls to the locations in prison api.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
This change will allow our service to make calls to the external locations in prisoner API for location based information e.g. rooms for video booking.

I will do each external API client separately to keep the number changes down.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/locationsinsideprison/LocationsInsidePrisonClient.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import reactor.core.publisher.Mono
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.locationsinsideprison.model.Location
+import java.util.UUID
+
+class LocationsInsidePrisonClient(private val locationsInsidePrisonApiWebClient: WebClient) {
+
+  fun getLocation(locationId: UUID): Location? = locationsInsidePrisonApiWebClient.get()
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/locationsinsideprison/LocationsInsidePrisonClient.kt:11]: Bit of an educated guess but seems likely we will want this one.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/wiremock/LocationsInsidePrisonApiMockServer.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  fun stubGetLocation(locationId: UUID = UUID.randomUUID(), prisonId: String = "MDI") {
+    stubFor(
+      get("/locations/$locationId").willReturn(
+        aResponse()
+          .withHeader("Content-Type", "application/json")
+          .withBody(
+            mapper.writeValueAsString(
+              Location(
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/wiremock/LocationsInsidePrisonApiMockServer.kt:38]: Bit of a random Location but will do for now.  Will probably add some factory classes to help deal with this later down the line.

File: helm_deploy/values-dev.yaml [hunk 3]
```diff
@@ -5,11 +5,12 @@ generic-service:
   replicaCount: 2
 
   ingress:
-    host: hmpps-book-a-video-link-api-dev.hmpps.service.justice.gov.uk
+    host: book-a-video-link-api-dev.prison.service.justice.gov.uk
```
Comment (timharrison-moj) [inline helm_deploy/values-dev.yaml:9]: Noticed this one earlier when looking at DB properties. Thank for fixing.

File: openapi-specs/locations-inside-prison-api.json [hunk 4]
```diff
@@ -0,0 +1 @@
...g"},"nullHandling":{"type":"string"},"ascending":{"type":"boolean"},"property":{"type":"string"},"ignoreCase":{"type":"boolean"}}},"Constant":{"required":["description","key"],"type":"object","properties":{"key":{"type":"string"},"description":{"type":"string"}}},"CompoundConstant":{"required":["description","key","values"],"type":"object","properties":{"key":{"type":"string"},"description":{"type":"string"},"values":{"type":"array","items":{"$ref":"#/components/schemas/Constant"}}}}},"securitySchemes":{"bearer-jwt":{"type":"http","name":"Authorization","in":"header","scheme":"bearer","bearerFormat":"JWT"},"hmpps-auth":{"type":"oauth2","flows":{"clientCredentials":{"tokenUrl":"https://sign-in-dev.hmpps.service.justice.gov.uk/auth/oauth/token","scopes":{"read":"Allows read of data","write":"Allows write of data"}},"authorizationCode":{"authorizationUrl":"https://sign-in-dev.hmpps.service.justice.gov.uk/auth/oauth/authorize","tokenUrl":"https://sign-in-dev.hmpps.service.justice.gov.uk/auth/oauth/token","scopes":{"read":"Allows read of data","write":"Allows write of data"}}},"openIdConnectUrl":"https://sign-in-dev.hmpps.service.justice.gov.uk/auth/.well-known/openid-configuration"}}}}
```
Comment (timharrison-moj) [inline openapi-specs/locations-inside-prison-api.json:1]: Do we normally check in these generated files? Couldn't remember.

**Review by timharrison-moj — APPROVED** [review]
Very good.

File: openapi-specs/locations-inside-prison-api.json (hunk 4 above)
Comment (matt-crowson-moj) [reply openapi-specs/locations-inside-prison-api.json:1]: We do yes.  The other option is to point at an actual URL e.g. the DEV URL.  One of the risks being if we do this things change and breaks our service without us knowing. Essentially it's a trade off.

**Pull request merged**